sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEEliminateLimits.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.adaptive
+
+import org.apache.spark.sql.catalyst.optimizer.EliminateLimitsBase
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+/**
+ * This rule runs in the AQE optimizer:
+ * [[LogicalQueryStage]] override the maxRows, so we can use the maxRows of logic plan to decide
+ * if we can eliminate limits. And we check if [[LogicalQueryStage]] is materialized at stats,
+ * if it is not materialized the maxRows is none.
+ */
+object AQEEliminateLimits extends EliminateLimitsBase {
```

Comment:
what's the difference between this and `EliminateLimits`?

Reply:
They are same. Just consider we don't need transformDownWithPruning in AQE Optimizer since it's batch only run once.

Reply:
that's too minor to justify creating a new class

Reply:
yea, seems can reuse `EliminateLimits`?

Reply:
OK, I'm fine to reuse the `EliminateLimits `